pkg/repositories/gormimpl/artifact.go

```diff
@@ -65,19 +66,20 @@ func (h *artifactRepo) Get(ctx context.Context, in models.ArtifactKey) (models.A
 		)
 
 	if result.Error != nil {
+		if result.Error.Error() == gorm.ErrRecordNotFound.Error() {
```

Comment:
I think this check is no longer necessary now that not found errors are returned in `result.Error` see https://github.com/flyteorg/datacatalog/blob/1a3b9195bb80ef23ea82eb25206e067afdf535af/pkg/repositories/errors/postgres.go#L30

here and in other places

Reply:
I think the error message returned will be different. In the old GORM version, we are returning

Reply:
true, the response code is the same. not really a fan of having to duplicate this code everywhere though seeing as we already have an error transformer. maybe we can pass in a formatted entity identifier to the error transformer so it can customize the error message